draft-ietf-oauth-browser-based-apps.md

```diff
@@ -367,6 +367,37 @@ the necessary CORS headers to enable the JavaScript code to make this POST reque
 from the domain on which the script is executing. (See {{cors}} for additional details.)
 
 
+### Tokens Securely Handled by a Service worker
+In this scenario, a [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) is responsible for obtaining tokens from the authorization server and passing them to the resource API.
+
+Service workers are inherently safe from XSS, because the browser APIs allowing to register one take an origin-constrained URL.
+They can thus be used as a safe store for tokens.
+
+Service workers can also be used to intercept calls from the frontend. As such, they can completely isolate calls to the authorization server from XSS attack surface.
+
+                                                             Resource               Authorization
+      User       Application    Service Worker                server                   server
+       |   browse     |               |                          |                        |
+       | ------------>|               |                          |          /authorize    |
+       |              |------------------------------------------------------------------->
+       |              |               |              redirect + authorization code        |
+       |              |               < - - - - - - - - - - - - - - - - - - - - - - - - - |
+       |              |               |              auth code                            |
+       |              |               | --------------->                      /token      |
+       |              |               | -------------------------------------------------->
+       |              |  rest call    | <- - - - - - - - - - - - - - - - - - - - - - - - -|
```

Comment:
it's not clear whether this "rest call" label is associated with the return from the authorization server or is the application calling the service worker. If the latter, then I don't think calling that a "rest call" is correct. Maybe just leave this label off?

Reply:
Replaced "rest call" with "resource request". Also added some further explanations.